### PR TITLE
TeachableAgent blog post

### PR DESCRIPTION
**Pull request by rickyloynd-microsoft**
## Why are these changes needed?
We need a blog post introducing `TeachableAgent` and explaining how to use it.

## Related issue number
None

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: website/blog/2023-10-26-TeachableAgent/index.mdx [hunk 1]
```diff
@@ -0,0 +1,390 @@
+---
+title: AutoGen's TeachableAgent
+authors: rickyloynd-microsoft
+tags: [LLM, teach]
+---
+
+![Teachable Agent Architecture](img/teachable-arch.png)
+
+**TL;DR:**
+* We introduce **TeachableAgent** (which uses by **TextAnalyzerAgent**) so that users can teach their LLM-based assistants new facts, preferences, and skills.
```
Comment (gagb) [inline website/blog/2023-10-26-TeachableAgent/index.mdx:10]: (which uses by **TextAnalyzerAgent**) --> awkward phrasing.

File: website/blog/2023-10-26-TeachableAgent/index.mdx [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+`TeachableAgent` addresses these limitations by persisting user teachings across chat boundaries in long-term memory implemented as a vector database. Memory is automatically saved to disk at the end of each chat, then loaded from disk at the start of the next. Instead of copying all of memory into the context window, which would eat up valuable space, individual memories (called memos) are retrieved into context as needed. This allows the user to teach frequently used facts and skills to the teachable agent just once, and have it recall them in later chats.
+
+In order to make effective decisions about memo storage and retrieval, `TeachableAgent` calls an instance of `TextAnalyzerAgent` to analyze pieces of text in various ways. This adds extra LLM calls involving a relatively small number of tokens, which can add a few seconds to the time a user waits for each response.
```
Comment (gagb) [inline website/blog/2023-10-26-TeachableAgent/index.mdx:19]: of `TextAnalyzerAgent` to analyze -->  of `TextAnalyzerAgent`, which is another AutoGen agent, to analyze

Comment (gagb) [inline website/blog/2023-10-26-TeachableAgent/index.mdx:19]: This adds extra LLM  --> Start with "Note that this adds"

**Review by gagb — COMMENTED** [review]
Left a few minor suggestions to improve phrasing. But otherwise looks good.

File: website/blog/2023-10-26-TeachableAgent/index.mdx (hunk 1 above)
Comment (rickyloynd-microsoft) [reply website/blog/2023-10-26-TeachableAgent/index.mdx]: fixed

File: website/blog/2023-10-26-TeachableAgent/index.mdx (hunk 2 above)
Comment (rickyloynd-microsoft) [reply website/blog/2023-10-26-TeachableAgent/index.mdx]: fixed

**Review by gagb — APPROVED** [review]

**Review by thinkall — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**